portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -45,33 +50,57 @@ private String convertToJsonUsingGson(List<String> messages) {
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Get comments in datastore, by most recent order at the top.
-    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    Query query = new Query("Comment").addSort(COMMENT_DATE, SortDirection.DESCENDING);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
-    // Iterate over all entities, get comment text.
-    List<String> comments = new ArrayList<>();
-    for (Entity entity : results.asIterable()) {
-      String commentText = (String) entity.getProperty("text");
-      comments.add(commentText);
-    }
+    // Extract limit on number of comments from query string (default is 5).
+    int limit = (request.getParameter("limit") == null) ?
+      5 : Integer.parseInt(request.getParameter("limit"));
+
+    // Iterate over all entities, get comment.
+    List<Comment> comments = getComments(results, limit);
 
     // Return comments in JSON format.
     response.setContentType("application/json;");
-    String json = convertToJsonUsingGson(comments);
+    String json = convertToJson(comments);
     response.getWriter().println(json);
   }
 
+  /**
+   * Get the comments from the datastore based on query in GET method.
```

Comment:
This comment is super readable (totally cool, it happens to the best of us), in particular the "on query in GET method.". If I could suggest a small rewrite?

"Get all Comment entities from the provided PreparedQuery. No more comments than the provided limit allows will be returned."

Reply:
Thanks for the suggested change, I put that in.